### PR TITLE
[QA][AC] add interface and override his method

### PR DESCRIPTION
**Pull request by Andrew-Chernenko**
add interface

### CODE REVIEW

File: src/main/java/Pages/CategoryPage.java [hunk 1]
```diff
@@ -13,4 +13,8 @@ public String returnUrl(){
         return driver.getCurrentUrl();
     }
 
+    @Override
+    public void waitLoadComponent() {
+
```
Comment (paulsinyakov) [inline src/main/java/Pages/CategoryPage.java]: add waiting for some elements

File: src/main/java/Pages/MainPage.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import org.openqa.selenium.support.Color;
 import org.openqa.selenium.support.FindBy;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
```
Comment (paulsinyakov) [inline src/main/java/Pages/MainPage.java]: unused import

File: src/main/java/Pages/MainPage.java [hunk 3]
```diff
@@ -39,12 +40,19 @@
     @FindBy(xpath = ".//div[@class='header-theme']//div[@class='switcher-toggle']")
     private WebElement checkBoxThemeSwitch;
 
+    @FindBy(xpath = ".//div[@class='banner__gradient']")
+    private WebElement slider;
+
+    @FindBy(xpath = ".//span[@class='language__ru active']")
+    private WebElement correctLanguage;
```
Comment (paulsinyakov) [inline src/main/java/Pages/MainPage.java]: pls rename this WebElement

File: src/main/java/Pages/MainPage.java [hunk 4]
```diff
@@ -56,7 +64,9 @@ public String getHeaderColor() {
 
     public void swapLanguageClick() {
         waitUtils.waitElementWithMiddleWaitAndClick(languageSwitch);
+        waitUtils.getElementWhenVisibleMiddleWait(correctLanguage);
         waitUtils.waitLoadingPageWithJavaScript();
+
```
Comment (paulsinyakov) [inline src/main/java/Pages/MainPage.java]: remove this linebreak

File: src/main/java/Pages/ProductListPage.java [hunk 5]
```diff
@@ -39,4 +39,9 @@ public String returnUrl(){
         List<WebElement> products = waitUtils.waitVisibilityAllElementsWithMiddleWait(productCart);
         return driver.getCurrentUrl();
     }
+
+    @Override
+    public void waitLoadComponent() {
+
```
Comment (paulsinyakov) [inline src/main/java/Pages/ProductListPage.java]: add waiting for some elements here

**Review by paulsinyakov — CHANGES_REQUESTED** [review]

File: src/main/java/Pages/CategoryPage.java [hunk 6]
```diff
@@ -1,16 +1,26 @@
 package Pages;
 import BasePage.BasePage;
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
 
 public class CategoryPage extends BasePage {
 
+    @FindBy(xpath = ".//div[@class='tiny-slider tiny-slider__content top-banner__slider']")
```
Comment (paulsinyakov) [inline src/main/java/Pages/CategoryPage.java]: use another element pls.
It's just an advertising.
https://prnt.sc/ul8l6z

**Review by paulsinyakov — CHANGES_REQUESTED** [review]

**Review by paulsinyakov — APPROVED** [review]

**Comment by Andrew-Chernenko** [comment]
Add interface

**Pull request merged**